spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/ModuleCommands.java

```diff
@@ -191,14 +195,45 @@ public Object getValue(int row, int column) {
 				int currentRow = row - 1;
 				if (mappings.get(key).size() > currentRow) {
 					return mappings.get(key).get(currentRow);
-				} else {
+				}
+				else {
 					return null;
 				}
 			}
 		};
 		return DataFlowTables.applyStyle(new TableBuilder(model)).build();
 	}
 
+	@CliCommand(value = IMPORT_APPS, help = "Register all apps listed in a properties file")
+	public String importFromResource(
+			@CliOption(mandatory = true,
+					key = {"", "uri"},
+					help = "URI for the properties file")
+			String uri,
+			@CliOption(key = "force",
+				help = "force update if any app already exists (only if not in use)",
+				specifiedDefaultValue = "true",
+				unspecifiedDefaultValue = "false")
+			boolean force) {
+		moduleOperations().importFromResource(uri, force);
+		return String.format("Successfully registered apps from '%s'", uri);
+	}
+
+	@CliCommand(value = REGISTER_APPS, help = "Register all apps provided in a map")
+	public String registerAll(
+			@CliOption(mandatory = true,
+					key = {"", "map"},
+					help = "apps as key/value pairs with key as 'type.name' and value as a URI")
+			Properties apps,
```

Comment:
I thought about this a bit too... using `\n` is not ideal in the shell (fine for the REST API though).

Actually makes me wonder if we need this command in the shell at all, since the other one (taking `--uri`) is already essentially the same thing as `--propertiesFile` (we could consider renaming that option in fact, since the latter is more explicit and matches what we do in other places, such as `stream deploy`)

Another option would be to allow a configurable `--delimiter`, with something more convenient as the default (maybe semicolon). I think it would need to be configurable in case the URIs being provided contain that character.